File: Maximum Nesting Depth of the Parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        int open_count = 0, maxx = 0;
        for (char &c: s) {
            if (c == '(')
                maxx = max(maxx, ++open_count);
            else if (c == ')')
                open_count--;
        }
        return maxx;
    }
};

File: Maximum Nesting Depth of the Parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        open_count = 0
        maxx = 0
        for c in s:
            if c == "(":
                open_count += 1
                maxx = max(maxx, open_count)
            elif c == ")":
                open_count -= 1
        return maxx
